test/src/main/java/org/apache/accumulo/test/metrics/MetricsIT.java

```diff
@@ -232,6 +258,186 @@ public void confirmMetricsPublished() throws Exception {
     cluster.stop();
   }
 
+  @Test
+  public void testFateExecutorMetrics() throws Exception {
+    // Tests metrics for Fate's thread pools. Tests that metrics are seen as expected, and config
+    // changes to the thread pools are accurately reflected in the metrics. This includes checking
+    // that old thread pool metrics are removed, new ones are created, size changes to thread
+    // pools are reflected, and the ops assigned and instance type tags are seen as expected
```

Comment:
Not a fan of how long and complex this test case is, but not much more simplification can be done without sacrificing test coverage